doc/contributing: add info about supportability and pre-requirements (#3707)
replicatedhq/kURL: CONTRIBUTING.md

# Contributing

We welcome contributions to kURL. We appreciate your time and help.

## Supportability

> kURL leverage on [kubeadin][kubeadmin]. Then, to test the project, you will need access to a `Linux/amd64`.
> You can check its [system requirements][kurl-system-requirements] for further info.
> 

Reviewer: ```suggestion
kURL leverages [kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/).
To test the project, you will need access to a `Linux/amd64` instance.
You can check [system requirements](https://kurl.sh/docs/install-with-kurl/system-requirements#supported-operating-systems) for further info.

```

## Pre-requirements

- npm _(i.e. for mac os: `brew install npm`)_
- [Golang](https://go.dev/doc/install) >= 1.19+
- Docker 

> **If your local environment is a Mac Os then:** ensure that you install `gnu-sed` and `md5sha1sum` to be able to run 
> the scripts with `brew install gnu-sed` and `brew install md5sha1sum`. Otherwise, you might face issues to run the scripts locally
> if you need.

Reviewer: ```suggestion
**If your local environment is OSX:** ensure that you install `gnu-sed` and `md5sha1sum` to be able to run the scripts.
These can be installed with `brew install gnu-sed` + `PATH="$(brew --prefix)/opt/gnu-sed/libexec/gnubin:$PATH"` and `brew install md5sha1sum`.
The apple-supplied `sed` does not support in-place replace, and OSX does not come with `md5sha1sum`
```

### QEMU on MacOS

> [kubeadmin]: https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/
> [kurl-system-requirements]: https://kurl.sh/docs/install-with-kurl/system-requirements#supported-operating-systems

Reviewer: ```suggestion
```
